src/main/java/frc/robot/subsystems/drive/Drivetrain.java

```diff
@@ -135,7 +143,7 @@ private void configureAutoBuilder() {
         this::getPose,
         this::resetOdometry,
         this::getSpeeds,
```

Comment:
I recommend you look at the implementation of this function and compare it to the example code from PathPlanner. I think there's a potential bug here with the speeds you're returning.

Reply:
use swerve drive kinematics to update every time method is called and don't rely on periodic not being overrun